src/main.rs

```diff
@@ -22,6 +24,43 @@ mod dvi;
 mod framebuffer;
 mod link;
 
+global_asm! {
+    "
+    .section .text
+    .align 4
+    data_cpy_table:
+     .word _scratch_x_source
+     .word _scratch_x_start
+     .word _scratch_x_end
+     .word _scratch_y_source
+     .word _scratch_y_start
+     .word _scratch_y_end
+     .word 0
+    .global __pre_init
+    .type __pre_init,%function
+    .thumb_func
+    __pre_init:
+     push {{r4, lr}}
+     ldr r4, =data_cpy_table
+
+    1:
+     ldmia r4!, {{r1-r3}}
+     cmp r1, #0
+     beq 2f
```

Comment:
My ARM Thumb assembly knowledge might be lacking, but this looks like a branch to a fixed location in memory `0x2f`, but that must be wrong. Does this have to do with the label `2`?